### PR TITLE
[To rel/0.12][ISSUE-4399] When non-root user returns a null value, return no permission error message

### PR DESCRIPTION
**Pull request by yifuzhou**
## Description
[ISSUE-4399](https://github.com/apache/iotdb/issues/4399)
TO reproduce:
user admin user:
SET STORAGE GROUP TO root.sg1;
CREATE TIMESERIES root.sg1.d1.s1 WITH DATATYPE=INT32, ENCODING=PLAIN;
CREATE TIMESERIES root.sg1.d1.s2 WITH DATATYPE=INT32, ENCODING=PLAIN;
INSERT INTO root.sg1.d1(timestamp, s1, s2) VALUES (0, -1, 1);
INSERT INTO root.sg1.d1(timestamp, s1, s2) VALUES (1, -2, 2);
INSERT INTO root.sg1.d1(timestamp, s1, s2) VALUES (2, -3, 3);

CREATE FUNCTION example AS "org.apache.iotdb.udf.UDTFExample";

SET STORAGE GROUP TO root.sg2;
CREATE TIMESERIES root.sg2.d1.s1 WITH DATATYPE=INT32, ENCODING=PLAIN;
CREATE TIMESERIES root.sg2.d1.s2 WITH DATATYPE=INT32, ENCODING=PLAIN;
INSERT INTO root.sg2.d1(timestamp, s1, s2) VALUES (0, -1, 1);
INSERT INTO root.sg2.d1(timestamp, s1, s2) VALUES (1, -2, 2);
INSERT INTO root.sg2.d1(timestamp, s1, s2) VALUES (2, -3, 3);

user a user A with only root.sg1 read permission:
pervious version:
image

after fixed it:
image

### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Comment by SteveYurongSu** [comment]
Hi, thanks for your contribution! Please take a look at the CI.

**Comment by yifuzhou** [comment]
> Hi, thanks for your contribution! Please take a look at the CI.

It looks weird that only one OS CI failed and fail to load the details.

**Comment by yifuzhou** [comment]
> > Hi, thanks for your contribution! Please take a look at the CI.
> 
> It looks weird that only one OS CI failed and fail to load the details.

I re-run it and it looks good! :)

**Review by wangchao316 — APPROVED** [review]
LGTM

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java [hunk 1]
```diff
@@ -130,6 +130,7 @@
 import org.apache.iotdb.tsfile.read.common.Path;
 import org.apache.iotdb.tsfile.read.query.dataset.QueryDataSet;
 
+import com.nimbusds.oauth2.sdk.util.CollectionUtils;
```
Comment (wangchao316) [inline server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java:133]: CollectionUtils need use commons-collections

**Review by wangchao316 — CHANGES_REQUESTED** [review]

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by SteveYurongSu — APPROVED** [review]
LGTM

**Pull request merged**